pkg/curatedpackages/regional_registry.go

```diff
@@ -22,28 +24,33 @@ var prodRegionalECRMap = map[string]string{
 	"us-east-2": "TODO.dkr.ecr.us-east-2.amazonaws.com",
 }
 
-// TestRegistryAccess test if the packageControllerClient has valid credential to access registry.
-func TestRegistryAccess(ctx context.Context, accessKey, secret, registry, region string) error {
-	cfg, err := config.LoadDefaultConfig(ctx,
-		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKey, secret, "")),
-		config.WithRegion(region),
-	)
-	if err != nil {
-		return err
-	}
+// RegistryAccessTester test if AWS credentials has valid permission to access an ECR registry.
+type RegistryAccessTester interface {
+	Test(ctx context.Context, accessKey, secret, region, awsConfig, registry string) error
+}
 
-	ecrClient := ecr.NewFromConfig(cfg)
-	out, err := ecrClient.GetAuthorizationToken(context.Background(), &ecr.GetAuthorizationTokenInput{})
+// DefaultRegistryAccessTester the default implementation of RegistryAccessTester.
+type DefaultRegistryAccessTester struct{}
+
+// Test if the AWS static credential or sharedConfig has valid permission to access an ECR registry.
+func (r *DefaultRegistryAccessTester) Test(ctx context.Context, accessKey, secret, region, awsConfig, registry string) (err error) {
+	authTokenProvider := &DefaultRegistryAuthTokenProvider{}
```

Comment:
what do you think about making this part of the struct? then it's a little easier to do dependency injection for testing?

Reply:
The disadvantage is that the field has to be specified clearly when creating the struct. Maybe the advantage doesn't worth the effort